internal/restic/node_windows.go

```diff
@@ -83,3 +107,208 @@ func (s statT) ctim() syscall.Timespec {
+
+// appendGenericAttribute appends a GenericAttribute to the node
+func (node *Node) appendGenericAttribute(genericAttribute GenericAttribute) {
+	if genericAttribute.Name != "" {
+		node.GenericAttributes = append(node.GenericAttributes, genericAttribute)
+	}
+}
+
+// getFileAttributes gets the value for the GenericAttribute TypeFileAttribute
+func getFileAttributes(fileattr uint32) (fileAttribute GenericAttribute) {
+	fileAttrData := UInt32ToBytes(fileattr)
+	fileAttribute = NewGenericAttribute(TypeFileAttribute, fileAttrData)
+	return fileAttribute
+}
+
+// UInt32ToBytes converts a uint32 value to a byte array
+func UInt32ToBytes(value uint32) (bytes []byte) {
+	bytes = make([]byte, 4)
+	binary.LittleEndian.PutUint32(bytes, value)
+	return bytes
+}
+
+// getCreationTime gets the value for the GenericAttribute TypeCreationTime in a windows specific time format.
+// The value is a 64-bit value representing the number of 100-nanosecond intervals since January 1, 1601 (UTC)
+// split into two 32-bit parts: the low-order DWORD and the high-order DWORD for efficiency and interoperability.
+// The low-order DWORD represents the number of 100-nanosecond intervals elapsed since January 1, 1601, modulo
+// 2^32. The high-order DWORD represents the number of times the low-order DWORD has overflowed.
+func getCreationTime(fi os.FileInfo, path string) (creationTimeAttribute GenericAttribute) {
+	attrib, success := fi.Sys().(*syscall.Win32FileAttributeData)
+	if success && attrib != nil {
+		var creationTime [8]byte
+		binary.LittleEndian.PutUint32(creationTime[0:4], attrib.CreationTime.LowDateTime)
+		binary.LittleEndian.PutUint32(creationTime[4:8], attrib.CreationTime.HighDateTime)
+		creationTimeAttribute = NewGenericAttribute(TypeCreationTime, creationTime[:])
+	} else {
+		debug.Log("Could not get create time for path: %s", path)
+	}
+	return creationTimeAttribute
+}
+
+// restoreGenericAttribute restores the generic attributes for windows like File Attributes,
+// Created time etc.
+func (attr GenericAttribute) restoreGenericAttribute(path string) error {
+	switch attr.Name {
+	case string(TypeFileAttribute):
+		return handleFileAttributes(path, attr.Value)
+	case string(TypeCreationTime):
+		return handleCreationTime(path, attr.Value)
+	}
+	handleUnknownGenericAttributeFound(attr.Name)
+	return nil
+}
+
+// handleFileAttributes gets the File Attributes from the data and sets them to the file/folder
+// at the specified path.
+func handleFileAttributes(path string, data []byte) (err error) {
+	attrs := binary.LittleEndian.Uint32(data)
+	pathPointer, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return err
+	}
+	err = fixEncryptionAttribute(path, attrs, pathPointer)
+	if err != nil {
+		debug.Log("Could not change encryption attribute for path: %s: %v", path, err)
+	}
+	return syscall.SetFileAttributes(pathPointer, attrs)
+}
+
+// fixEncryptionAttribute checks if a file needs to be marked encrypted and is not already encrypted, it sets
```

Comment:
Fixed backup and restore of FILE_ATTRIBUTE_ENCRYPTED in windows.

Reply:
Encrypted attribute can only be set when a file is being newly created. When calling `SetFileAttribute` with file attributes including encrypted attribute, windows simply ignores the encrypted (`FILE_ATTRIBUTE_ENCRYPTED`) attribute and applies the rest.
The other way to handle this would be to delete a file if it exists and recreate it with `FILE_ATTRIBUTE_ENCRYPTED` if needed in fileswriter, which was making fileswriter very complex and more so when adding ADS functionality to it.